### ISSUE TITLE
question about "#include "instantiate_templates.hpp" & duplicate symbol

### ISSUE THREAD

**Issue by alaingdl** [gnudatalanguage/gdl#1046]
On OSX with GCC (gcc-11 & c++-11 provided by Brew) I have a very large number of _duplicate symbol_ and the link fails.

```
[...]
duplicate symbol __ZTS5Data_I6SpDIntE in:
    CMakeFiles/gdl.dir/datatypes.cpp.o
    CMakeFiles/gdl.dir/basic_op_div.cpp.o
[...]
duplicate symbol __ZTS5Data_I8SpDFloatE in:
    CMakeFiles/gdl.dir/datatypes.cpp.o
    CMakeFiles/gdl.dir/default_io.cpp.o
[...]
```

1. It seems that all the files (cpp) concerned are containing `#include "instantiate_templates.hpp"`
2. Could someone explains why we do need that, and at the end of those files ?
(and not at the beginning ?)

List : datatypes.cpp,  all files basic_op_*.cpp, objects.cpp, ofmt.cpp, ifmt.cpp ...
(grep -ni instantiate src/*pp)

**Comment by GillesDuvert** [comment]
see #734
The duplication is indeed due to several .cpp compiling and creating their version of the same symbols. And linkers may or not accept that. We know that these symbols are the same functions, the duplication is, er, harmless. Could yo try to add
`--allow-multiple-definition`
as a switch to the linker (`ld`)

**Comment by GillesDuvert** [comment]
--allow-multiple-definition ... is not supported on gcc à la homebrew (???)

**Comment by GillesDuvert** [comment]
closed by #1776
